src/replication.c

```diff
@@ -3626,6 +3626,14 @@ void syncWithPrimary(connection *conn) {
         err = sendCommand(conn, "REPLCONF", "version", VALKEY_VERSION, NULL);
         if (err) goto write_error;
 
+        /* Inform the primary of our (replica) node name. */
+        if (server.cluster_enabled) {
+            char *argv[] = {"CLIENT", "SETNAME", server.cluster->myself->name};
```

Comment:
I don't understand why this isn't a replconf. It seems like this should be more specific information than just setting it as the name.

Reply:
Replconf needs some more work and a new field in the client struct, etc. Why is it bad to reuse the setname?

Psync and replconf are internal undocumented commands so I don't see any risk. Is it about code complexity?

Reply:
I thought about reusing `REPLCONF` as well but since what we need a replica's identity, I feel that `CLIENT SETNAME` works too. Ideally though I would love to see us deduce the name automatically on the server side. One idea could be looking up the remote address against all the replica nodes after detecting a replica client.

Reply:
Neither https://valkey.io/commands/replconf/ nor https://valkey.io/commands/psync/ are technically undocumented, I agree they aren't very well documented but they are part of the protocol and people build tooling around it. 

My other concern is now show up with a weird name that weren't there before, so people might take dependencies on it. Do we want to forever behind the replica name (we could instead put something more human readable there)? It's a bit of a binding decision, and would rather have flexibility.

Reply:
REPLCONF has no documented arguments. The page just says "Usage: REPLCONF".

Reply:
If we document it, I think it will mostly be confusing to users.

Users might think they can send it to the primary  like with standalone FAILOVER TO, but it's not replicated so it doesn't work.

I agree though that even if we don't document it, we probably need to keep it around.

Reply:
> My other concern is now show up with a weird name that wasn't there before, so people might take dependencies on it.

Good point. The replicas show up in CLIENT LIST.

Reply:
> The replicas show up in CLIENT LIST.

I use SETNAME on purpose. I would like to show it in the client list so that we can quickly locate the node and its associated node id in the replication part (not the cluster part). But it’s not that important, sorry i did not mention this impact at the first time.